src/main/java/core/basesyntax/service/impl/WriterImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.Writer;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class WriterImpl implements Writer {
+    private static final String REPORT_HEAD_LINE = "fruit,quantity";
+    private static final char COMMA = ',';
+
+    @Override
+    public void createReport(String reportPath) {
+        File report = new File(reportPath);
+        StringBuilder reportText = new StringBuilder(REPORT_HEAD_LINE);
+        for (Map.Entry<String, Fruit> fruit : Storage.fruits.entrySet()) {
```

Comment:
Access to the Storage.fruits should be encapsulated within the Storage class. Consider creating a method in the Storage class to get the entries, rather than accessing the field directly. This would adhere to the encapsulation principle.